### DISCUSSION TITLE
What makes a crate compatible with the risc0 zkvm

### DISCUSSION THREAD

**Discussion by 0xfinetuned** [risc0/risc0#1292] [Q&A] [answered]
I'm trying to use secp256k1 crate on the zkvm guest and it doesn't work because it's not compatible. 

I'm very curious about what makes a crate compatible so I can make an implementation that's compatible and can be used in the zkvm guest.

I'm aware that you can't use a random number generator in the zkvm guest and secp256k1 uses one, it this the only reason why it's not supported or I'm missing something else ?

**Comment by pdg744** [comment]
There are various reasons. Many crates don't specify RISC-V as a target, or they only support 64 bit versions of RISC-V. 

You can see the error logs for various failing crates in the [crate validation report](https://reports.risczero.com/crates-validation). We also have a [`rust-guest-workarounds`](https://github.com/risc0/risc0/issues?q=is%3Aissue+label%3A%22rust+guest+workarounds%22+) tag that may provide some insight into some of the other possible issues and workarounds. 

I've opened #1509 as a place for conversation about secp256k1 specifically.

**Comment by flaub** [comment, marked as answer]
This crate should already be patched here: https://github.com/risc0/RustCrypto-elliptic-curves/tree/risczero
